packages/dropdown-react/src/Dropdown.tsx

```diff
@@ -69,76 +80,63 @@ export const Dropdown = ({
-                className="jkl-dropdown__value"
-                data-testid="jkl-dropdown__value"
-                aria-haspopup="listbox"
-            >
-                {hasSelectedValue ? selectedValue : defaultPrompt}
-            </button>
-            <CoreToggle
-                id={listId}
-                role="listbox"
-                className="jkl-dropdown__core-toggle"
-                popup={label}
-                hidden={!dropdownIsShown}
-                onToggle={onToggle}
-                onToggleSelect={onToggleSelect}
-                aria-activedescendant={hasSelectedValue && `${listId}__${lower(selectedValue)}`}
-            >
-                <ul
-                    className="jkl-dropdown__option-wrapper"
-                    data-testid="jkl-dropdown__option-wrapper"
-                    tabIndex={-1}
-                    ref={listRef}
+        <div data-testid="jkl-dropdown" className={componentClassName}>
+            <span className={labelClassName}>{label}</span>
+            <div className="jkl-dropdown__outer-wrapper">
+                <button
+                    type="button"
+                    className="jkl-dropdown__value"
+                    data-testid="jkl-dropdown__value"
+                    aria-haspopup="listbox"
                 >
-                    {items.map((item) => (
-                        <li key={item}>
-                            <button
-                                type="button"
-                                id={`${listId}__${lower(item)}`}
-                                className="jkl-dropdown__option"
-                                data-testid="jkl-dropdown__option"
-                                aria-selected={item === selectedValue}
-                                role="option"
-                            >
-                                {item}
-                            </button>
-                        </li>
-                    ))}
-                </ul>
-            </CoreToggle>
-
-            <span className="jkl-dropdown__chevron" />
+                    {hasSelectedValue ? displayedValue : defaultPrompt}
+                </button>
+                <CoreToggle
+                    id={listId}
+                    role="listbox"
+                    className="jkl-dropdown__core-toggle"
+                    popup={label}
+                    hidden={!dropdownIsShown}
+                    onToggle={onToggle}
+                    onToggleSelect={onToggleSelect}
+                    aria-activedescendant={hasSelectedValue && `${listId}__${lower(selectedValue)}`}
+                >
+                    <ul
+                        className="jkl-dropdown__option-wrapper"
+                        data-testid="jkl-dropdown__option-wrapper"
+                        tabIndex={-1}
+                        ref={listRef}
+                    >
+                        {items.map((item) => {
+                            item = getSelectValuePairFrom(item);
```

Comment:
Personally I'm not a fan of reassigning function parameters but I'd rather create a new variabel. Specially here, since the type changes (for some cases) and its har do know what exactly `item` is.

Reply:
You can also use it in a map chain:           
```
     {items.map(getSelectValuePairFrom).map((item) => (
                    <option data-testid="jkl-dropdown__option" key={item.value} value={item.value}>
                        {item.label}
                    </option>
                ))}
```